### ISSUE TITLE
Bug in modules/payloads/singles/linux/x64/shell_find_port.rb

### ISSUE THREAD

**Issue by CCob** [rapid7/metasploit-framework#11908]
## Steps to reproduce

How'd you do it?

1. Generate linux/x64/shell_find_port payload with `msfvenom -p linux/x64/shell_find_port -f python LHOST=127.0.0.1 CPORT=6666`

2. Run shell code where RSP contents is all 0x00 

The current shell code expects the [rsp] value to contain a valid sockt_len, if the value at [rsp] at the time of execute is < 0x10, the syscall does succeed, but does not fill in [rsi] with valid data as the buffer length is too small.  A `mov [rdx], 0x10` or equivalent instruction after 'lea rdx, [rsp]' should do the trick

```
            xor rdi,rdi
            xor rbx,rbx
            mov bl,0x14
            sub rsp,rbx
            lea rdx,[rsp]   <---- Assumption here that *[rsp] >= 0x10
            lea rsi,[rsp+4]
          find_port:
            push 0x34     ; getpeername
            pop rax
            syscall
            inc rdi
            cmp word [rsi+2],0x4142
            jne find_port
            dec rdi
            push 2
            pop rsi
```

## Expected behaviour

TCP shell spawned on existing socket

## Current behaviour

`find_port` loop stuck in endless loop as existing socket cannot be found

### Metasploit version

Latest master

### I installed Metasploit with:
- [x] Omnibus installer (nightly)

### OS

What OS are you running Metasploit on?

Ubuntu

**Comment by sempervictus** [comment]
Thanks for the detailed analysis and bug report. Given that you've not only identified the problem but also propose a solution, would you like to submit a pull request with the relevant line added? Seems the work you put in to track this down merits immortalizing in the commit log :)

**Comment by CCob** [comment]
Sure.  I'll see what I can do

**Comment by CCob** [comment]
Tested, seems to work where it was failing with me previously.

**Comment by space-r7** [comment]
Fixed in #11923
